File: src/c-leetcode/first/Code_104_MaxDepthOfBT.c
#include "../TreeNode.h"
#include "../TreeNode.c"
#include <stdio.h>

/**
 * 二叉树最大深度
 * 
 * @since 2020-8-20 Thursday 19:07
 */
int max(int i, int j);

int maxDepth(struct TreeNode *root)
{
    if (root == NULL)
        return 0;
    return max(maxDepth(root->left), maxDepth(root->right)) + 1;
}

int max(int i, int j)
{
    return i > j ? i : j;
}

int main(int argc, char *argv[])
{
    printf("%d\n", maxDepth(binaryTree()));
    printf("%d\n", maxDepth(binarySearchTree()));
}

File: src/leetcode/third/Code_104_MaxDepthOfBT.java
package leetcode.third;

import leetcode.tool.TreeNode;

/**
 * 二叉树的最大深度
 *
 * @since 2019-12-15 Sunday
 */
public class Code_104_MaxDepthOfBT {
    static int maxDepth(TreeNode root) {
        if (root == null) return 0;
        return Math.max(maxDepth(root.right) + 1, maxDepth(root.left) + 1);
    }

    public static void main(String[] args) {
        TreeNode root = new TreeNode(1);
        root.left = new TreeNode(1);
        root.right = new TreeNode(2);
        root.left.left = new TreeNode(3);
        root.left.left.left = new TreeNode(3);
        System.out.println(maxDepth(root));
    }
}
